lib/rules/object-curly-newline.js

```diff
@@ -144,6 +144,9 @@ module.exports = {
                 )
             );
 
+            const allTokens = sourceCode.getTokens(node, { includeComments: true });
+            const hasComments = allTokens.some(token => astUtils.isCommentToken(token));
```

Comment:
That change did not work. I'll be happy to make other changes if you'd like. Also, I updated the test files as suggested.